### ISSUE TITLE
Consider slightly different volume structure

### ISSUE THREAD

**Issue by rmunn** [sillsdev/languageforge-lexbox#885]
While researching #884, I came across [this advice](https://www.postgresql.org/docs/16/creating-cluster.html#CREATING-CLUSTER-MOUNT-POINTS) in the PostgreSQL manual (emphasis mine):

> Many installations create their database clusters on file systems (volumes) other than the machine's “root” volume. If you choose to do this, **it is not advisable to try to use the secondary volume's topmost directory (mount point) as the data directory**. Best practice is to create a directory within the mount-point directory that is owned by the PostgreSQL user, and then create the data directory within that. This avoids permissions problems, particularly for operations such as pg_upgrade, and it also ensures clean failures if the secondary volume is taken offline.

This is actually exactly the situation we've been running into. We currently mount our data volume into `/var/hg/repos` on hgweb, and the data volume contains the folders `a`, `b`, `c` and so on. Which means that we need the volume root to be mounted with owner `www-data` and group `www-data` in order to be able to create new folders, and we've had to do some initContainer tricks to make that possible. (Because by default on Linux, mount points are owned by the `root` user, so we have to run an initContainer after the mount to change the permissions of the volume root).

However, what if we changed it so that the volume contains one folder named `repos`, and is mounted onto `/var/hg`? Then the necessary permissions on the `repos` folder would be **stored in the volume data itself** and would not need to be re-created every time. That would go a long way towards making our permissions issues a lot less gnarly.

**Comment by hahn-kev** [comment]
this would require moving all the repos in our volume if I understand correctly no?

**Comment by rmunn** [comment]
Yes, it would, but as a one-time thing. And we should be able to do something with symlinks so that no code changes would be required.
